ampd/src/grpc/blockchain_service.rs

```diff
@@ -120,27 +138,30 @@ mod tests {
 
     async fn setup(
         mock_event_sub: MockEventSub,
-        mut mock_cosmos_client: MockCosmosClient,
+        mut broadcaster_mock_cosmos_client: MockCosmosClient,
+        mock_cosmos_client: MockCosmosClient,
```

Comment:
why two separate mocks?

Reply:
One is for the broadcaster, and one is for the grpc server itself. In production, they are just a clone of each other. In tests, I wanna ensure separate behaviour.